packages/beacon-node/src/sync/unknownBlock.ts

```diff
@@ -147,19 +150,34 @@ export class UnknownBlockSync {
     this.addUnknownBlock(parentBlockRootHex, peerIdStr);
   }
 
-  private addUnknownBlock(blockRootHex: RootHex, peerIdStr?: string): void {
+  private addUnknownBlock(blockInputOrRootHex: RootHex | BlockInput, peerIdStr?: string): void {
+    let blockRootHex;
+    let blockInput: BlockInput | null;
+
+    if (typeof blockInputOrRootHex === "string") {
+      blockRootHex = blockInputOrRootHex;
+      blockInput = null;
+    } else {
+      const {block} = blockInputOrRootHex;
+      blockRootHex = toHexString(this.config.getForkTypes(block.message.slot).BeaconBlock.hashTreeRoot(block.message));
+      blockInput = blockInputOrRootHex;
+    }
+
     let pendingBlock = this.pendingBlocks.get(blockRootHex);
     if (!pendingBlock) {
       pendingBlock = {
         blockRootHex,
         parentBlockRootHex: null,
-        blockInput: null,
+        blockInput,
         peerIdStrs: new Set(),
         status: PendingBlockStatus.pending,
         downloadAttempts: 0,
       };
       this.pendingBlocks.set(blockRootHex, pendingBlock);
-      this.logger.verbose("Added unknown block to pendingBlocks", {root: blockRootHex});
```

Comment:
may want to separate to `Added unknown block to pendingBlocks` and `Added missing blobs to pendingBlocks`

also `pendingBlocks` maybe confusing now, change to `pendingBlockInputs`?